docs/source/developer/mvp_guidelines.rst

```diff
@@ -0,0 +1,207 @@
+
+.. _MVP_Guidelines:
+
+=====================================
+Model-View-Presenter (MVP) Guidelines
+=====================================
+
+Overview
+========
+
+The Model-View-Presenter (MVP) pattern is a software design pattern
+that separates the structure of an application into three main components:
+Model, View, and Presenter. This separation enhances modularity, testability,
+and maintainability of the codebase.
+
+Components
+===========
+
+1. **Model:**
+   The Model represents the application's data and algorithm logic. It is
+   responsible for retrieving and manipulating data, as well as notifying the
+   Presenter of any changes. The Model is independent of both the View and the
+   Presenter.
+
+2. **View:**
+   The View is responsible for presenting the data to the user and receiving
+   user input. It remains passive and delegates user interactions to the Presenter.
```

Comment:
there is some logic in the view. can we rephrase into:
It does not contain the full application logic or something like that?